browser-sign-in-kotlin/app/src/main/java/com/okta/browser/MainActivity.kt

```diff
@@ -171,21 +177,62 @@ class MainActivity : AppCompatActivity(), SharedPreferences.OnSharedPreferenceCh
         webAuthClient?.signOutOfOkta(this)
     }
 
+    private fun signInSuccess() {
+        showMessage(getString(R.string.authorized))
+        supportFragmentManager.beginTransaction().replace(
+            R.id.fragment,
+            AuthorizedFragment.newInstance(customSignIn)
+        ).commit()
+    }
+
+    private fun signInError(msg: String?, exception: AuthorizationException?) {
+        network_progress.hide()
+        showMessage(msg ?: getString(R.string.unknown))
+        Log.d(tag, "onError: ", exception)
+    }
+
     private fun signIn(
         customSignIn: Boolean,
         hintOrUsername: String? = null,
         password: String? = null
     ) {
-        networkCallInProgress()
         if (customSignIn) {
-            hintOrUsername.isNullOrEmpty().and(password.isNullOrEmpty())
-                .run { showMessage(getString(R.string.invalid_input)) }
-
+            hintOrUsername.isNullOrEmpty().or(password.isNullOrEmpty())
```

Comment:
This whole construct because we except null inputs? Is there no easier way to write this?

Reply:
yes the password can be null. I'll just put in empty string if it is null.